### ISSUE TITLE
Using outputs from reusable workflow call causes a type error

### ISSUE THREAD

**Issue by rhysd** [rhysd/actionlint#121]
### Input

```yaml
name: Call a reusable workflow and use its outputs

on:
  workflow_dispatch:

jobs:
  job1:
    uses: octo-org/example-repo/.github/workflows/called-workflow.yml@v1

  job2:
    runs-on: ubuntu-latest
    needs: job1
    steps:
      - run: echo ${{ needs.job1.outputs.firstword }} ${{ needs.job1.outputs.secondword }}
```

obtained from [the example in the doc](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-outputs-from-a-reusable-workflow)

### Expected results

No error occurs

### Actual results

```
test.yaml:14:23: property "firstword" is not defined in object type {} [expression]
   |
14 |       - run: echo ${{ needs.job1.outputs.firstword }} ${{ needs.job1.outputs.secondword }}
   |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
test.yaml:14:59: property "secondword" is not defined in object type {} [expression]
   |
14 |       - run: echo ${{ needs.job1.outputs.firstword }} ${{ needs.job1.outputs.secondword }}
   |                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Since outputs of reusable workflow call are not defined in `jobs.<job_id>` section in the same workflow, `needs.<job_id>.outputs` is defined as `{}` incorrectly.

### Environment

- actionlint v1.6.9
- Go 1.17.8
- macOS 11
